lib/cmd/common/initlocal.js

```diff
@@ -256,59 +266,39 @@ function doInitLocal(projectId, appId, cb) {
   log.silly('Getting app container from url: ' + fhreq.getFeedHenryUrl() + ", domain: " + fhc.curTarget, 'get container');
 
   // check valid appid
-  common.readApp(projectId, appId, function (err) {
-    if (err) {
-      return cb(err);
-    }
-    installFhDbLocally(function (err) {
-      if (err) {
-        return cb(err);
-      }
-      createLocalDirectory(localDirectory, function (err) {
+  return async.waterfall([
+    common.readApp.bind(this, projectId, appId),
+    installFhDbLocally,
+    createLocalDirectory.bind(this, localDirectory),
+    checkApplicationJS.bind(this, applicationJS),
+    common.getAppContainer.bind(this, appId),
```

Comment:
That's the refactoring needed for using `async.waterfall`. It takes an array of functions that either receive the return of the previous ones as params and the callback as the last one. Originally they were using variables available from the closure as params to direct calls.

I use `bind` to create a version of the functions that only receive the callback or params that are actually return values of the previous functions. I've tested this manually and actually found a bug on running the command on my local vm, but the refactoring works the same as the original code, and the bug also happens in upstream.